src/smartHandler.ts

```diff
@@ -184,21 +191,36 @@ export class SMARTHandler implements Authorization {
     }
 
     async getSearchFilterBasedOnIdentity(request: GetSearchFilterBasedOnIdentityRequest): Promise<SearchFilter[]> {
+        console.log('inside getSearchFilterBasedOnIdentity function.');
         const references: Set<string> = new Set();
         const ids: Set<string> = new Set();
-        const { fhirUserObject, patientLaunchContext, usableScopes } = request.userIdentity;
+        const { fhirUserObject, patientLaunchContext, usableScopes, patientOrgsClaim } = request.userIdentity;
         const fhirServiceBaseUrl = request.fhirServiceBaseUrl ?? this.apiUrl;
 
         if (hasSystemAccess(usableScopes, '')) {
             return [];
         }
 
+        // this exist if the scope starts with user i.e., practitioner or patient
```

Comment:
This comment is throwing me off a little.  Isn't the fhirUserObject created because of the existence of a fhirUser claim, and not because of certain scopes beginning with user?

Reply:
yes, you are correct. Might be some leftover comments. Will remove it. In our case FhirUser could be either Patient or User.